app/models/points/participation.rb

```diff
@@ -0,0 +1,40 @@
+module Points
+  class Participation
+    def calculate(game, row)
+      @game = game
+      @row = row
+
+      points_for_participation.tap do |points|
+        return 0 if current >= max || week_current >= week_max
+        return (max - current) if (current + points) >= max
+        return (week_max - week_current) if (week_current + points) >= week_max
+      end
+    end
```

Comment:
If anything, I overuse `tap`! I have a passionate dislike of dangling return variables, so
```
def do_the_thing
  thing = whatever
  thing.do_it!
  thing
end
```
reads much more nicely to me as:
```
def do_the_thing
  whatever.tap do |thing|
    thing.do_it!
  end
end
```